src/bin_heap.py

```diff
@@ -0,0 +1,64 @@
+class BinHeap(object):
+    """Class created for  max binary heap."""
+
+    def __init__(self, iterable=None):
+        """Initialize max heap."""
+        self._heap = []
+        if iterable is not None:
+            for item in iterable:
+                self._heap.append(item)
+                self._heapify()
+
+    def _heapify(self):
+        """Heapify the list."""
+        if not self._heap:
+            return
+        i = len(self._heap) - 1
+        parent = (i - 1) // 2
+        while True:
+            if self._heap[i] > self._heap[parent]:
+                self._heap[i], self._heap[parent] = self._heap[parent], self._heap[i]
```

Comment:
You repeat this operation many times in the program -- you could DRY up your code and make it more maintainable by abstracting this operation into a `swap` function.